src/hooks/useListModules.ts

```diff
@@ -21,6 +30,32 @@ const useListModules = () => {
     [dispatch, setRequestId]
   );
 
+  const listDeliverables = useCallback(() => {
+    const dispatched = dispatch(requestListDeliverables({ locale: activeLocale }));
+    setDeliverablesRequestId(dispatched.requestId);
+  }, [dispatch, setDeliverablesRequestId]);
+
+  useEffect(() => {
+    if (listModulesResponse && allDeliverablesResponse.status === 'success') {
+      const deliverables = allDeliverablesResponse.data;
+      const _deliverablesByModuleId: Record<number, ListDeliverablesElementWithOverdue[]> = {};
+      deliverables?.forEach((_deliverable) => {
+        const moduleId = _deliverable.moduleId;
+        if (_deliverablesByModuleId[moduleId]) {
+          if (
+            _deliverablesByModuleId[moduleId].every((existingDeliverable) => existingDeliverable.id !== _deliverable.id)
+          ) {
+            [..._deliverablesByModuleId[moduleId], _deliverable];
+          }
+        } else {
+          _deliverablesByModuleId[moduleId] = [_deliverable];
+        }
+      });
+
+      setDeliverablesByModuleId(_deliverablesByModuleId);
```

Comment:
This only shows one deliverable which I think is due to some overwriting.

But also, we have a `fetchDeliverables` (https://github.com/terraware/terraware-server/blob/e4d2afd302085be741681a801c6c9804c9210c91/src/main/kotlin/com/terraformation/backend/accelerator/db/DeliverableStore.kt#L59), it is probably worth exposing this via an API so you have a list of deliverables by modules! I can get started on this API today :)

Reply:
Yes, I think having it in the BE will be useful. I added in the FE because I need it! Let me know if you are adding it in the BE, or if not I will review why is this returning one deliverable, I think it worked, but I may have changed something and broke it

Reply:
Also in the modules section, there is another modules table where we show the deliverables quantity and the cohort quantity. Maybe we can also add that in the BE?

Reply:
I think we only have search for project modules and deliverables. Probably easiest to expose an API since we already have to store method. I will submit a PR soon!

Reply:
Thanks!

Reply:
https://github.com/terraware/terraware-server/pull/2470

Reply:
I couldn't find an easy way to fit it into our existing APIs. you may need to write a cached selector around it and dispatch per module. Sort of a pain :(

Reply:
I think it is more complex to do a request for each module, than just getting all deliverables and group them by module. So I fixed what I previously did (it was a small change). Let me know what you think

Reply:
Sounds good! I think this is fine as it is!

If you do want to use the new endpoint, you can do something similar to updateMany at the redux level with a Promise.all. But this looks like it is working well already!